unittests/CMakeLists.txt

```diff
@@ -1,4 +1,31 @@
 
+if(SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER)
+  if((NOT "${CMAKE_C_COMPILER_ID}" STREQUAL "Clang") AND
+     (NOT "${CMAKE_C_COMPILER_ID}" STREQUAL "AppleClang"))
+    message(FATAL_ERROR "Building the swift runtime is not supported with ${CMAKE_C_COMPILER_ID}. Use the just-built clang instead.")
```

Comment:
This is the same thing that `stdlib/CMakeLists.txt` does, so it's wrong in both places if so. (Which could be!) These tests have to use whatever stdlib uses to avoid the calling convention mismatch.